src/ucp/core/ucp_ep.c

```diff
@@ -3339,6 +3319,139 @@ static ucs_status_t ucp_ep_query_sockaddr(ucp_ep_h ep, ucp_ep_attr_t *attr)
+        return status;
+    }
+
+    return ucs_sockaddr_copy(sockaddr, (struct sockaddr*)&attr.remote_sockaddr);
+}
+
+static ucs_status_t ucp_ep_vfs_read_local_addr(ucp_ep_h ep,
+                                               ucs_string_buffer_t *strb,
+                                               struct sockaddr *sockaddr)
+{
+    ucp_ep_attr_t attr;
+    ucs_status_t status;
+
+    status = ucp_ep_vfs_query_sockaddr(ep, &attr,
+                                       UCP_EP_ATTR_FIELD_LOCAL_SOCKADDR, strb);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    return ucs_sockaddr_copy(sockaddr, (struct sockaddr*)&attr.local_sockaddr);
+}
+
+static void ucp_ep_vfs_read_remote_addr_ip(void *obj, ucs_string_buffer_t *strb,
+                                           void *arg_ptr, uint64_t arg_u64)
+{
+    struct sockaddr sockaddr;
+
+    if (ucp_ep_vfs_read_remote_addr((ucp_ep_h)obj, strb, &sockaddr) == UCS_OK) {
+        ucp_vfs_read_ip(&sockaddr, strb);
+    }
+}
+
+static void ucp_ep_vfs_read_remote_addr_port(void *obj,
+                                             ucs_string_buffer_t *strb,
+                                             void *arg_ptr, uint64_t arg_u64)
+{
+    struct sockaddr sockaddr;
+
+    if (ucp_ep_vfs_read_remote_addr((ucp_ep_h)obj, strb, &sockaddr) == UCS_OK) {
+        ucp_vfs_read_port(&sockaddr, strb);
+    }
+}
+
+static void ucp_ep_vfs_read_local_addr_ip(void *obj, ucs_string_buffer_t *strb,
+                                          void *arg_ptr, uint64_t arg_u64)
+{
+    struct sockaddr sockaddr;
+
+    if (ucp_ep_vfs_read_local_addr((ucp_ep_h)obj, strb, &sockaddr) == UCS_OK) {
+        ucp_vfs_read_ip(&sockaddr, strb);
+    }
+}
+
+static void ucp_ep_vfs_read_local_addr_port(void *obj,
+                                            ucs_string_buffer_t *strb,
+                                            void *arg_ptr, uint64_t arg_u64)
+{
+    struct sockaddr sockaddr;
+
+    if (ucp_ep_vfs_read_local_addr((ucp_ep_h)obj, strb, &sockaddr) == UCS_OK) {
+        ucp_vfs_read_port(&sockaddr, strb);
+    }
+}
+
+void ucp_ep_vfs_init(ucp_ep_h ep)
+{
+    ucp_err_handling_mode_t err_mode;
+    ucp_ep_attr_t attr;
+    struct sockaddr *sockaddr;
+
+#if ENABLE_DEBUG_DATA
+    ucs_vfs_obj_add_dir(ep->worker, ep, "ep/%s", ep->name);
+    ucs_vfs_obj_add_ro_file(ep, ucs_vfs_show_memory_address, NULL, 0,
+                            "memory_address");
+#else
+    ucs_vfs_obj_add_dir(ep->worker, ep, "ep/%p", ep);
+#endif
+
+    ucs_vfs_obj_add_ro_file(ep, ucp_ep_vfs_show_peer_name, NULL, 0,
+                            "peer_name");
+
+    err_mode = ucp_ep_config(ep)->key.err_mode;
+    ucs_vfs_obj_add_ro_file(ep, ucs_vfs_show_primitive,
+                            (void*)ucp_err_handling_mode_names[err_mode],
+                            UCS_VFS_TYPE_STRING, "error_mode");
+
+    attr.field_mask = UCP_EP_ATTR_FIELD_REMOTE_SOCKADDR;
+    if (ucp_ep_query_sockaddr(ep, &attr) == UCS_OK) {
```

Comment:
even if the endpoint is not connected yet, but was created in client/server mode, we should create the sockaddr files
and looks like ucp_ep_query_sockaddr() will fail if the endpoint is not connected yet